src/qt/bitcoin.cpp

```diff
@@ -308,7 +308,7 @@ void BitcoinApplication::startThread()
 
     /*  communication to and from thread */
     connect(&m_executor.value(), &InitExecutor::initializeResult, this, &BitcoinApplication::initializeResult);
-    connect(&m_executor.value(), &InitExecutor::shutdownResult, this, &QCoreApplication::quit);
+    connect(&m_executor.value(), &InitExecutor::shutdownResult, [] { QCoreApplication::exit(0); });
```

Comment:
The signal `shutdownResult` is emitted from executor's thread and this results in calling `exit()` in that thread because the connection lacks a context. But `exit()` must be called in the main thread: https://doc-snapshots.qt.io/qt6-dev/qcoreapplication.html#exit. Maybe:
```cpp
connect(&m_executor.value(), &InitExecutor::shutdownResult, this, [] {
    QCoreApplication::exit(0);
});

Reply:
Why? `QCoreApplication::exit()` is a static member function.

Reply:
Yes, but according to the doc

> Note also that this function is not thread-safe. It should be called only from the main thread

and by using https://doc.qt.io/qt-5/qobject.html#connect-5 we ensure that lambda is called on the right thread.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/586#issuecomment-1133648314).